smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/httpResponse/bindingTraits/HttpResponseTraitWithHttpPayload.kt

```diff
@@ -25,62 +25,76 @@ class HttpResponseTraitWithHttpPayload(
+                    if (target.isEnum) {
+                        writer.write("self.\$L = \$L(rawValue: output)", memberName, symbol)
+                    } else {
                         writer.write("self.\$L = output", memberName)
                     }
-                    writer.indent()
-                    writer.write("self.\$L = nil", memberName).closeBlock("}")
                 }
-                ShapeType.STRING -> {
-                    writer.openBlock("if let output = \$N(data: data, encoding: .utf8) {", "} else {", SwiftTypes.String) {
-                        if (target.isEnum) {
-                            writer.write("self.\$L = \$L(rawValue: output)", memberName, symbol)
-                        } else {
-                            writer.write("self.\$L = output", memberName)
-                        }
-                    }
-                    writer.indent()
-                    writer.write("self.\$L = nil", memberName).closeBlock("}")
+                writer.indent()
+                writer.write("self.\$L = nil", memberName).closeBlock("}")
+            }
+            ShapeType.ENUM -> {
+                writer.openBlock("if let data = try httpResponse.body.toData(), let output = \$N(data: data, encoding: .utf8) {", "} else {", SwiftTypes.String) {
+                    writer.write("self.\$L = \$L(rawValue: output)", memberName, symbol)
                 }
-                ShapeType.ENUM -> {
-                    writer.openBlock("if let output = \$N(data: data, encoding: .utf8) {", "} else {", SwiftTypes.String) {
-                        writer.write("self.\$L = \$L(rawValue: output)", memberName, symbol)
-                    }
-                    writer.indent()
-                    writer.write("self.\$L = nil", memberName).closeBlock("}")
+                writer.indent()
+                writer.write("self.\$L = nil", memberName).closeBlock("}")
+            }
+            ShapeType.BLOB -> {
+                writer.write("switch httpResponse.body {")
+                    .write("case .data(let data):")
+                    .indent()
+                if (isBinaryStream) {
+                    writer.write("self.\$L = .data(data)", memberName)
+                } else {
+                    writer.write("self.\$L = data", memberName)
                 }
-                ShapeType.BLOB -> {
-                    if (isBinaryStream) {
-                        writer.write("self.\$L = ByteStream.from(data: data)", memberName)
-                    } else {
-                        writer.write("self.\$L = data", memberName)
-                    }
+
+                // For binary streams, we need to set the member to the stream directly.
+                // this allows us to stream the data directly to the user
+                // without having to buffer it in memory.
```

Comment:
This comment mentions binary streams, yet a few liens above it appears we handle binary streams by setting body to `.data(data)`. Do we handle binary streams in multiple ways?